chirp/drivers/kg935g.py

```diff
@@ -430,6 +412,35 @@
 # MRT 1.2 correct spelling of Wouxon
 
 
+def str2callid(val):
+    """ Convert caller id strings from callid2str.
+    """
+    ascii2bin = "0123456789"
+    s = str(val).strip()
+    LOG.debug("val = %s" % val)
+    LOG.debug("s = %s" % s)
+    if len(s) < 3 or len(s) > 6:
+        raise InvalidValueError(
+            "Caller ID must be at least 3 and no more than 6 digits")
+    if s[0] == '0':
+        raise InvalidValueError(
+            "First digit of a Caller ID cannot be a zero '0'")
+    blk = bytearray()
+    for c in s:
+        if c not in ascii2bin:
+            raise InvalidValueError(
+                "Caller ID must be all digits 0x%x" % c)
+        b = (0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9)[int(c)]
+        blk.append(b)
+    if len(blk) < 6:
+        blk.append(0xf)  # EOL a short ID
+    if len(blk) < 6:
+        for i in range(0, (6 - len(blk))):
+            blk.append(0xf0)
```

Comment:
Surely you can combine L435-L439 right? However, if you use a `bytes` instead, you can do:
```
blk.ljust(6, 0xF0)
```
Which makes sure to fill up to 6 positions with a byte.

Reply:
So on this one, instead of bytearray above, try:
```
blk = bytes()
for c in s:
  blk.append(asc2bin.index(c))
return blk.ljust(6, b'\xF0')
```

Reply:
so this one has 2 different fills.  
end of data byte is 0xF while the rest will be filled with 0xF0

Reply:
probably should be cleaner and use 0x0F and 0xF0 to not add confusion

Reply:
Ah, okay I missed that. Just leave it then, that's fine.